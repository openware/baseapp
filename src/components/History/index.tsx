import { CellData, Table } from '@openware/components';
import classnames from 'classnames';
import * as React from 'react';

export interface HistoryProps {
    /**
     * List of history data
     */
    data: CellData[][];
    /**
     * List of headers for history table
     */
    headers?: string[];
}

export class History extends React.PureComponent<HistoryProps> {
    private defaultHeaders = ['Time', 'Action', 'Price', 'Amount', 'Total'];
    private title = 'Trades History';

    public render() {
        const { headers = this.defaultHeaders } = this.props;
        const tableData = this.props.data.map(row => row.map(this.mapRows));
        return (
            <Table
                data={tableData}
                header={headers}
                titleComponent={this.title}
            />
        );
    }

    public renderAction(actionType: string) {
        const action = actionType ? actionType.toLowerCase() : actionType;
        const className = classnames('cr-history-action', {
            'cr-history-action--buy': action === 'bid',
            'cr-history-action--sell': action === 'ask',
        });

        return <span className={className}>{action}</span>;
    }

    private mapRows = (cell: CellData, index: number) => {
        const { headers = this.defaultHeaders } = this.props;
        const actionIndex = headers.findIndex(header => header === 'Action');
        return index === actionIndex ? this.renderAction(cell as string) : cell;
    }
}
