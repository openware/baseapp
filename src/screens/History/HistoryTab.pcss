.pg-history-tab {
  align-items: flex-start !important;
  background: var(--body-background-color);
  border-radius: var(--gap);
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin-bottom: 24px !important;
  margin-top: calc(var(--gap) * 2) !important;
  max-width: 920px !important;
  padding: 0 !important;

  .cr-pagination {
    text-align: center;
  }

  .pg-history-tab__tabs-content {
    align-items: flex-start;
    width: 100%;

    .cr-tab-panel__navigation-container {
      background: var(--tab-panel-background-color);
      border-bottom: 1px solid var(--tab-panel-border-line-color);
      border-top-left-radius: calc(var(--gap) * 0.66);
      border-top-right-radius: calc(var(--gap) * 0.66);
      margin-bottom: 0;

      &-navigation {
        .cr-tab {
          padding-left: calc(var(--gap) * 7);
          padding-right: calc(var(--gap) * 7);
        }
      }
    }
  }

  .pg-history-elem-empty {
    align-items: center;
    display: flex;
    justify-content: center;
    min-height: 50vh;
  }

  .pg-history-elem-loading {
    align-items: center;
    display: flex;
    justify-content: center;
  }

  .pg-history-elem {
    padding-top: var(--gap);

    &__empty {
      color: var(--secondary-contrast-cta-color);
      text-align: center;
    }

    &__pagination {
      padding: var(--gap) calc(var(--gap) * 2);

      p {
        margin-bottom: 0;
      }
    }

    .text-center {
      align-items: center;
      bottom: 0;
      display: flex;
      justify-content: center;
      left: 0;
      position: absolute;
      right: 0;
      top: 0;
    }

    .cr-table-container {
      padding: 0 calc(var(--gap) * 2);

      .cr-table-header__content {
        display: none;
      }

      .cr-table {
        background: transparent;
        border-collapse: collapse;

        &__head {
          margin: 10px;

          &-row {
            border-bottom: 1px solid var(--table-border-color);

            th {
              background-color: var(--table-background-color);
              color: var(--table-text-color);
              font-family: "Roboto", "Helvetica", "Arial", sans-serif;
              font-size: 12px !important;
              font-weight: 500;
              padding: calc(var(--gap) * 2.7) 0 calc(var(--gap) * 2.7) !important;
              position: static;
              text-align: right;
              width: calc(var(--big-gap) * 2);

              &:first-child {
                margin-left: 10px;
                text-align: left;
              }

              &:nth-child(7) {
                white-space: nowrap;
              }
            }
          }
        }

        &__body {
          background-color: var(--table-background-color);

          tr {
            background-color: var(--table-background-color);
            border-bottom: 1px solid var(--table-border-color);

            td {
              background-color: var(--table-background-color);
              color: var(--table-text-color);
              font-size: 13px !important;
              font-weight: 400;
              margin-bottom: calc(var(--gap) * 1.66) !important;
              margin-top: calc(var(--gap) * 1.66) !important;
              padding: calc(var(--gap) * 2.7) 0 calc(var(--gap) * 2.7) !important;
              text-align: right;

              &:first-child {
                text-align: left;
              }

              &:last-child {
                &::first-letter {
                  text-transform: uppercase;
                }
              }

              a {
                color: var(--table-text-color);
                text-decoration: none;
              }
            }
          }
        }
      }
    }
  }

  @media screen and (max-width: 768px) {
    .pg-history-elem {
      &__hide {
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }

    .cr-table-container {
      .cr-table {
        overflow: scroll;
      }
    }

    .cr-tab-panel {
      &__navigation-container {
        overflow: hidden;
      }
    }

    .cr-tab {
      position: relative;
      right: 7%;
      top: 50%;
      transform: translateY(-50%);
      width: 1%;

      &__pointer {
        left: 40% !important;
        width: 80% !important;
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .pg-history-elem {
    &__hide {
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }

  .cr-table-container {
    .cr-table {
      overflow: scroll;
    }
  }
}
