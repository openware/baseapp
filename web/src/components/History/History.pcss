.cr-history {
  &-action {
    background: transparent;

    &--buy {
      color: var(--history-action-buy);
    }

    &--sell {
      color: var(--history-action-sell);
    }
  }
}
