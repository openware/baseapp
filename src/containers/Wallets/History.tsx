import * as React from 'react';
import {
    InjectedIntlProps,
    injectIntl,
} from 'react-intl';
import { connect, MapDispatchToPropsFunction } from 'react-redux';
import { Decimal } from '../../components/Decimal';
import { History, WalletItemProps, Pagination } from '../../components';
import { localeDate } from '../../helpers';
import {
    currenciesFetch,
    Currency,
    fetchHistory,
    resetHistory,
    RootState,
    selectCurrencies,
    selectCurrentPage,
    selectFirstElemIndex,
    selectFullHistory,
    selectHistory,
    selectHistoryLoading,
    selectLastElemIndex,
    selectNextPageExists,
    selectPageCount,
    selectWallets,
    WalletHistoryList,
} from '../../modules';
import { FailIcon } from './FailIcon';
import { SucceedIcon } from './SucceedIcon';

export interface HistoryProps {
    label: string;
    type: string;
    currency: string;
}

export interface ReduxProps {
    currencies: Currency[];
    list: WalletHistoryList;
    wallets: WalletItemProps[];
    fetching: boolean;
    fullHistory: number;
    page: number;
    pageCount: number;
    firstElemIndex: number;
    lastElemIndex: number;
    nextPageExists: boolean;
}

interface DispatchProps {
    fetchCurrencies: typeof currenciesFetch;
    fetchHistory: typeof fetchHistory;
    resetHistory: typeof resetHistory;
}

export type Props = HistoryProps & ReduxProps & DispatchProps & InjectedIntlProps;

export class WalletTable extends React.Component<Props> {
    public componentDidMount() {
        const {
            currencies,
            currency,
            type,
        } = this.props;
        this.props.fetchHistory({ page: 0, currency, type, limit: 6 });

        if (currencies.length === 0) {
            this.props.fetchCurrencies();
        }
    }

    public componentWillReceiveProps(nextProps) {
        const {
            currencies,
            currency,
            type,
        } = this.props;
        if (nextProps.currency !== currency || nextProps.type !== type) {
            this.props.resetHistory();
            this.props.fetchHistory({ page: 0, currency: nextProps.currency, type, limit: 6 });
        }

        if (nextProps.currencies.length === 0 && nextProps.currencies !== currencies) {
            this.props.fetchCurrencies();
        }
    }

    public componentWillUnmount() {
        this.props.resetHistory();
    }

    public render() {
        const { label, list, fullHistory, firstElemIndex, lastElemIndex, page, nextPageExists } = this.props;

        if (!list.length) {
            return null;
        }
        return (
            <div className="pg-history-elem__wallet">
                <div className="pg-history-elem__label">
                    {this.props.intl.formatMessage({ id: `page.body.history.${label}` })}
                </div>
                <History headers={this.getHeaders(label)} data={this.retrieveData(list)} />
                <Pagination
                    firstElemIndex={firstElemIndex}
                    lastElemIndex={lastElemIndex}
                    total={fullHistory}
                    page={page}
                    nextPageExists={nextPageExists}
                    onClickPrevPage={this.onClickPrevPage}
                    onClickNextPage={this.onClickNextPage}
                />
            </div>
        );
    }

    private getHeaders = (label: string) => [
        this.props.intl.formatMessage({ id: `page.body.history.${label}.header.date` }),
        this.props.intl.formatMessage({ id: `page.body.history.${label}.header.status` }),
        this.props.intl.formatMessage({ id: `page.body.history.${label}.header.amount` }),
    ];

    private onClickPrevPage = () => {
        const { page, type, currency } = this.props;
        this.props.fetchHistory({ page: Number(page) - 1, currency, type, limit: 6 });
    };

    private onClickNextPage = () => {
        const { page, type, currency } = this.props;
        this.props.fetchHistory({ page: Number(page) + 1, currency, type, limit: 6 });
    };

    private retrieveData = list => {
        const {
            currency,
            currencies,
            intl,
            type,
            wallets,
        } = this.props;
        const { fixed } = wallets.find(w => w.currency === currency) || { fixed: 8 };
        if (list.length === 0) {
            return [[intl.formatMessage({ id: 'page.noDataToShow' }), '', '']];
        }
        return list.sort((a, b) => {
            return localeDate(a.created_at, 'fullDate') > localeDate(b.created_at, 'fullDate') ? -1 : 1;
        }).map((item, index) => {
            const amount = 'amount' in item ? Number(item.amount) : Number(item.price) * Number(item.volume);
            const confirmations = type === 'deposits' && item.confirmations;
            const itemCurrency = currencies && currencies.find(cur => cur.id === currency);
            const minConfirmations = itemCurrency && itemCurrency.min_confirmations;
            const state = 'state' in item ? this.formatTxState(item.state, confirmations, minConfirmations) : '';
            return [
                localeDate(item.created_at, 'fullDate'),
                state,
                <Decimal key={index} fixed={fixed}>{amount}</Decimal>,
            ];
        });
    };

    private formatTxState = (tx: string, confirmations?: number, minConfirmations?: number) => {
        const statusMapping = {
            succeed: <SucceedIcon />,
            failed: <FailIcon />,
            accepted: <SucceedIcon />,
            collected: <SucceedIcon />,
            canceled: <FailIcon />,
            rejected: <FailIcon />,
            processing: this.props.intl.formatMessage({ id: 'page.body.wallets.table.pending' }),
            prepared: this.props.intl.formatMessage({ id: 'page.body.wallets.table.pending' }),
            submitted: (confirmations !== undefined && minConfirmations !== undefined) ? (
                `${confirmations}/${minConfirmations}`
            ) : (
                this.props.intl.formatMessage({ id: 'page.body.wallets.table.pending' })
            ),
            skipped: <SucceedIcon />,
        };
        return statusMapping[tx];
    };
}


export const mapStateToProps = (state: RootState): ReduxProps => ({
    currencies: selectCurrencies(state),
    list: selectHistory(state),
    wallets: selectWallets(state),
    fetching: selectHistoryLoading(state),
    fullHistory: selectFullHistory(state),
    page: selectCurrentPage(state),
    pageCount: selectPageCount(state, 6),
    firstElemIndex: selectFirstElemIndex(state, 6),
    lastElemIndex: selectLastElemIndex(state, 6),
    nextPageExists: selectNextPageExists(state, 6),
});

export const mapDispatchToProps: MapDispatchToPropsFunction<DispatchProps, {}> =
    dispatch => ({
        fetchCurrencies: () => dispatch(currenciesFetch()),
        fetchHistory: params => dispatch(fetchHistory(params)),
        resetHistory: () => dispatch(resetHistory()),
    });

export const WalletHistory = injectIntl(connect(mapStateToProps, mapDispatchToProps)(WalletTable));
