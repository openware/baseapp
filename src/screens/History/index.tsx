import * as React from 'react';
import {
    injectIntl,
} from 'react-intl';
import {
    connect,
    MapDispatchToPropsFunction,
} from 'react-redux';
import { compose } from 'redux';
import { IntlProps } from '../../';
import { TabPanel } from '../../components';
import { HistoryElement } from '../../containers/HistoryElement';
import { setDocumentTitle } from '../../helpers';
import {
    fetchHistory,
    marketsFetch,
    resetHistory,
    walletsFetch,
} from '../../modules';

interface DispatchProps {
    resetHistory: typeof resetHistory;
    fetchMarkets: typeof marketsFetch;
    fetchWallets: typeof walletsFetch;
    fetchHistory: typeof fetchHistory;
}

type Props = DispatchProps & IntlProps;

interface State {
    tab: string;
    currentTabIndex: number;
}

class History extends React.Component<Props, State> {
    public state = {
        tab: 'deposits',
        currentTabIndex: 0,
    };

    public tabMapping = ['deposits', 'withdraws', 'trades'];

    public componentDidMount() {
        setDocumentTitle('History');
        this.props.fetchMarkets();
        this.props.fetchWallets();
    }

    public componentWillUnmount() {
        this.props.resetHistory();
    }

    public render() {
        return (
            <div className="pg-history-tab pg-container">
                <div className="pg-history-tab__tabs-content">
                    <TabPanel
                        panels={this.renderTabs()}
                        onTabChange={this.handleMakeRequest}
                        currentTabIndex={this.state.currentTabIndex}
                        onCurrentTabChange={this.onCurrentTabChange}
                    />
                </div>
            </div>
        );
    }

    private onCurrentTabChange = index => this.setState({ currentTabIndex: index });

    private handleMakeRequest = (index: number) => {
        if (this.state.tab === this.tabMapping[index]) {
            return;
        }
        this.props.resetHistory();
        this.setState({ tab: this.tabMapping[index] });
    };

    private renderTabs = () => {
        const { tab } = this.state;

        return [
            {
                content: tab === 'deposits' ? <HistoryElement type="deposits" /> : null,
                label: this.props.intl.formatMessage({id: 'page.body.history.deposit'}),
            },
            {
                content: tab === 'withdraws' ? <HistoryElement type="withdraws" /> : null,
                label: this.props.intl.formatMessage({id: 'page.body.history.withdraw'}),
            },
            {
                content: tab === 'trades' ? <HistoryElement type="trades" /> : null,
                label: this.props.intl.formatMessage({id: 'page.body.history.trade'}),
            },
        ];
    };
}

const mapDispatchToProps: MapDispatchToPropsFunction<DispatchProps, {}> = dispatch => ({
    fetchMarkets: () => dispatch(marketsFetch()),
    fetchWallets: () => dispatch(walletsFetch()),
    fetchHistory: payload => dispatch(fetchHistory(payload)),
    resetHistory: () => dispatch(resetHistory()),
});

export const HistoryScreen = compose(
    injectIntl,
    connect(null, mapDispatchToProps),
)(History) as React.ComponentClass;
