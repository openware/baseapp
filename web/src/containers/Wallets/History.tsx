import * as React from 'react';
import { injectIntl } from 'react-intl';
import { connect, MapDispatchToPropsFunction } from 'react-redux';
import { compose } from 'redux';
import { IntlProps } from '../../';
import { History, Pagination } from '../../components';
import { Decimal } from '../../components/Decimal';
import { localeDate } from '../../helpers';
import { LinkIcon } from '../../assets/images/LinkIcon';
import {
    Currency,
    fetchHistory,
    resetHistory,
    RootState,
    selectCurrencies,
    selectCurrentPage,
    selectFirstElemIndex,
    selectHistory,
    selectHistoryLoading,
    selectLastElemIndex,
    selectNextPageExists,
    selectWallets,
    selectWithdrawSuccess,
    Wallet,
    WalletHistoryList,
} from '../../modules';
import { FailIcon } from './FailIcon';
import { SucceedIcon } from './SucceedIcon';

export interface HistoryProps {
    label: string;
    type: string;
    currency: string;
}

export interface ReduxProps {
    currencies: Currency[];
    list: WalletHistoryList;
    wallets: Wallet[];
    fetching: boolean;
    page: number;
    firstElemIndex: number;
    lastElemIndex: number;
    nextPageExists: boolean;
    withdrawSuccess?: boolean;
}

interface DispatchProps {
    fetchHistory: typeof fetchHistory;
    resetHistory: typeof resetHistory;
}

const rowsPerPage = 6;

export type Props = HistoryProps & ReduxProps & DispatchProps & IntlProps;

export class WalletTable extends React.Component<Props> {
    public componentDidMount() {
        const { currency, type } = this.props;

        this.props.fetchHistory({ page: 0, currency, type, limit: 6 });
    }

    public componentWillReceiveProps(nextProps: Props) {
        const { currency, type } = this.props;

        if (nextProps.currency !== currency || nextProps.type !== type) {
            this.props.resetHistory();
            this.props.fetchHistory({ page: 0, currency: nextProps.currency, type, limit: rowsPerPage });
        }
    }

    public componentWillUnmount() {
        this.props.resetHistory();
    }

    public render() {
        const { label, list, firstElemIndex, lastElemIndex, page, nextPageExists } = this.props;

        if (!list.length) {
            return null;
        }

        return (
            <div className="pg-history-elem__wallet">
                <div className="pg-history-elem__label">
                    {this.props.intl.formatMessage({ id: `page.body.history.${label}` })}
                </div>
                <History headers={this.getHeaders(label)} data={this.retrieveData(list)} />
                <Pagination
                    firstElemIndex={firstElemIndex}
                    lastElemIndex={lastElemIndex}
                    page={page}
                    nextPageExists={nextPageExists}
                    onClickPrevPage={this.onClickPrevPage}
                    onClickNextPage={this.onClickNextPage}
                />
            </div>
        );
    }

    private getHeaders = (label: string) => [
        this.props.intl.formatMessage({ id: `page.body.history.${label}.header.date` }),
        this.props.intl.formatMessage({ id: `page.body.history.${label}.header.status` }),
        this.props.intl.formatMessage({ id: `page.body.history.${label}.header.amount` }),
    ];

    private onClickPrevPage = () => {
        const { page, type, currency } = this.props;
        this.props.fetchHistory({ page: Number(page) - 1, currency, type, limit: rowsPerPage });
    };

    private onClickNextPage = () => {
        const { page, type, currency } = this.props;
        this.props.fetchHistory({ page: Number(page) + 1, currency, type, limit: rowsPerPage });
    };

    private retrieveData = list => {
        const {
            currency,
            currencies,
            type,
            wallets,
        } = this.props;
        const { fixed } = wallets.find(w => w.currency === currency) || { fixed: 8 };

        if (!list.length) {
            return [[]];
        }

        return list.sort((a, b) => {
            return localeDate(a.created_at, 'fullDate') > localeDate(b.created_at, 'fullDate') ? -1 : 1;
        }).map((item, index) => {
            const blockchainLink = this.getBlockchainLink(currency, item.txid, item.rid);
            const amount = 'amount' in item ? Number(item.amount) : Number(item.price) * Number(item.volume);
            const confirmations = type === 'deposits' && item.confirmations;
            const itemCurrency = currencies && currencies.find(cur => cur.id === currency);
            const blockchainCurrency = itemCurrency?.networks?.find(blockchain_cur => blockchain_cur.blockchain_key === item.blockchain_key);
            const minConfirmations = blockchainCurrency?.min_confirmations;
            const state = 'state' in item ? this.formatTxState(item.state, confirmations, minConfirmations) : '';

            return [
                localeDate(item.created_at, 'fullDate'),
                <div className="pg-history-elem__hide" key={item.txid || item.rid}>
                    <a href={blockchainLink} target="_blank" rel="noopener noreferrer">
                        {item.txid || item.rid}
                    </a>
                    <LinkIcon className="pg-history-elem__link" />
                </div>,
                state,
                <Decimal key={index} fixed={fixed} thousSep=",">{amount}</Decimal>,
            ];
        });
    };

    private getBlockchainLink = (currency: string, txid: string, blockchainKey: string, rid?: string) => {
        const { wallets } = this.props;
        const currencyInfo = wallets?.find(wallet => wallet.currency === currency);
        const blockchainCurrency = currencyInfo?.networks?.find(blockchain_cur => blockchain_cur.blockchain_key === blockchainKey);

        if (currencyInfo) {
            if (txid && blockchainCurrency?.explorerTransaction) {
                return blockchainCurrency.explorerTransaction.replace('#{txid}', txid);
            }
            if (rid && blockchainCurrency?.explorerAddress) {
                return blockchainCurrency.explorerAddress.replace('#{address}', rid);
            }
        }

        return '';
    };

    private formatTxState = (tx: string, confirmations?: number | string, minConfirmations?: number) => {
        const statusMapping = {
            succeed: <SucceedIcon />,
            failed: <FailIcon />,
            accepted: <SucceedIcon />,
            collected: <SucceedIcon />,
            canceled: <FailIcon />,
            rejected: <FailIcon />,
            processing: this.props.intl.formatMessage({ id: 'page.body.wallets.table.pending' }),
            prepared: this.props.intl.formatMessage({ id: 'page.body.wallets.table.pending' }),
            submitted: (minConfirmations && confirmations && confirmations !== 'N/A') ? (
                `${confirmations}/${minConfirmations}`
            ) : (
                this.props.intl.formatMessage({ id: 'page.body.wallets.table.pending' })
            ),
            skipped: <SucceedIcon />,
        };

        return statusMapping[tx];
    };
}


export const mapStateToProps = (state: RootState): ReduxProps => ({
    currencies: selectCurrencies(state),
    list: selectHistory(state),
    wallets: selectWallets(state),
    fetching: selectHistoryLoading(state),
    page: selectCurrentPage(state),
    firstElemIndex: selectFirstElemIndex(state, 6),
    lastElemIndex: selectLastElemIndex(state, 6),
    nextPageExists: selectNextPageExists(state, 6),
    withdrawSuccess: selectWithdrawSuccess(state),
});

export const mapDispatchToProps: MapDispatchToPropsFunction<DispatchProps, {}> =
    dispatch => ({
        fetchHistory: params => dispatch(fetchHistory(params)),
        resetHistory: () => dispatch(resetHistory()),
    });

export const WalletHistory = compose(
  connect(mapStateToProps, mapDispatchToProps),
  injectIntl,
)(WalletTable) as any;
