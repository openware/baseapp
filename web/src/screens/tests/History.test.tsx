import { shallow } from 'enzyme';
import React from 'react';
import { TestComponentWrapper } from 'lib/test';
import { IntlProps } from '../../';
import { HistoryScreen } from '../History';

const setup = (props: Partial<IntlProps> = {}) =>
    shallow(
        <TestComponentWrapper>
            <HistoryScreen />
        </TestComponentWrapper>
    );

describe('HistoryScreen', () => {
    it('should render', () => {
        const wrapper = setup().render();
        expect(wrapper).toMatchSnapshot();
    });
});
