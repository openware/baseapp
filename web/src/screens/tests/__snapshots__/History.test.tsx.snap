// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryScreen should render 1`] = `
<div
  class="pg-history-tab pg-container"
>
  <div
    class="pg-history-tab__tabs-content"
  >
    <div
      class="cr-tab-panel"
    >
      <div
        class="cr-tab-panel__navigation-container draggable-container"
      >
        <div
          class="cr-tab-panel__navigation-container-navigation"
          role="tablist"
        >
          <div
            class="cr-tab cr-tab__active"
            role="tab"
            tabindex="0"
          >
            Deposit History
            <span
              class="cr-tab__pointer"
            />
          </div>
          <div
            class="cr-tab"
            role="tab"
            tabindex="1"
          >
            Withdrawal History
          </div>
          <div
            class="cr-tab"
            role="tab"
            tabindex="2"
          >
            Trade History
          </div>
          <div
            class="cr-tab"
            role="tab"
            tabindex="3"
          >
            Transfer History
          </div>
        </div>
      </div>
      <div
        class="cr-tab-content cr-tab-content__active"
      >
        <div
          class="pg-history-elem pg-history-elem-empty"
        >
          <p
            class="pg-history-elem__empty"
          >
            There is no data to show
          </p>
        </div>
      </div>
      <div
        class="cr-tab-content"
      />
      <div
        class="cr-tab-content"
      />
      <div
        class="cr-tab-content"
      />
    </div>
  </div>
</div>
`;
